samples/graph-meeting-notification/csharp/MeetingNotification/Bots/MeetingNotificationBot.cs

```diff
@@ -0,0 +1,109 @@
+    using System.Threading;
+    using System.Threading.Tasks;
+    using System.Web;
+
+    public class MeetingNotificationBot : TeamsActivityHandler
+    {
+        /// <summary>
+        /// Logs the information of bot hanadler.
+        /// </summary>
+        protected readonly ILogger logger;
+
+        /// <summary>
+        /// Store the coversation reference of meeting.
+        /// </summary>
+        private readonly ConcurrentDictionary<string, ConversationReference> conversationReferences;
+
+        /// <summary>
+        /// Manages the subscriptions created.
+        /// </summary>
+        private readonly SubscriptionManager subscriptionManager;
+
+        public MeetingNotificationBot(ILogger<MeetingNotificationBot> logger, ConcurrentDictionary<string, ConversationReference> conversationReferences, SubscriptionManager subscriptionManager)
+        {
+            this.logger = logger;
+            this.conversationReferences = conversationReferences;
+            this.subscriptionManager = subscriptionManager;
+        }
+
+        /// <summary>
+        /// Adds the coversation reference in dictionary.
+        /// </summary>
+        /// <param name="activity">Bot activity.</param>
+        /// <param name="meetingUrl">Url of meeting passed as key.</param>
+        private void AddConversationReference(Activity activity, string meetingUrl)
+        {
+            var conversationReference = activity.GetConversationReference();
+            this.conversationReferences.AddOrUpdate(meetingUrl, conversationReference, (key, newValue) => conversationReference);
+        }
+
+        protected override async Task OnInstallationUpdateAddAsync(ITurnContext<IInstallationUpdateActivity> turnContext, CancellationToken cancellationToken)
+        {
+            try
+            {
+                var meetingInfo = await TeamsInfo.GetMeetingInfoAsync(turnContext, null, cancellationToken);
+                var decodedMeetingJoinUrl = HttpUtility.UrlDecode(meetingInfo.Details.JoinUrl.ToString());
+                AddConversationReference(turnContext.Activity as Activity, decodedMeetingJoinUrl);
+                await this.subscriptionManager.InitializeAllSubscription(meetingInfo.Details.JoinUrl.ToString());
+
+                await turnContext.SendActivityAsync(MessageFactory.Attachment(NotificationCardHelper.GetSubscriptionInitializedCard(true)), cancellationToken);
+
+                while (!cancellationToken.IsCancellationRequested)
+                {
+                    await Task.Delay(TimeSpan.FromMinutes(15), cancellationToken).ConfigureAwait(false);
+                    await this.subscriptionManager.CheckSubscriptions().ConfigureAwait(false); ;
+                }
+            }
+            catch (Exception ex)
+            {
+                this.logger.LogWarning(ex.Message);
+
+                if (ex.Message == "The meetingId can only be null if turnContext is within the scope of a MS Teams Meeting.")
```

Comment:
This comparison may not the correct way to compare.
error message may change and it can fail the condition.
Can't we check the exception type  or code ?

Reply:
We are not getting any error code or type specific to outside installation we can only compare ex.Message